native/ops_cub/device_reduce.cu

```diff
@@ -2,15 +2,30 @@
 
 namespace device_reduce {
 
-using namespace goldilocks;
-using namespace memory;
-
 struct offset_iterator {
   const int offset;
   const int stride;
+
   DEVICE_FORCEINLINE int operator[](const int idx) const { return offset + idx * stride; }
 };
 
+} // namespace device_reduce
+
+#if CUB_VERSION >= 200300
+namespace cuda::std {
+
+template <> struct iterator_traits<::device_reduce::offset_iterator> {
+  typedef int value_type;
```

Comment:
For the record: you specialized `struct iterator_traits` to provide what you need for your local type, but the problem is i have no idea what _else_ `struct iterator_traits` is supposed to implement or provide. It's probably fine but we should still prefer using upstream cccl when it's fixed.